app/services/PenaltiesService.scala

```diff
@@ -157,9 +164,10 @@ class PenaltiesService @Inject()(config: FrontendAppConfig,
 
   //SELECT SCHEME
   def penaltySchemes(year: String, psaId: String)
-                            (implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Seq[PenaltySchemes]] =
+                    (implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Seq[PenaltySchemes]] =
     for {
       penalties <- fsConnector.getPsaFS(psaId)
+      _ <- fiCacheConnector.save(Json.obj("pstrs" -> penalties.map(_.pstr)))
```

Comment:
I feel you may need to filter out duplicates here (you need to just store the unassociated ones) - when I try it locally I get lots of duplicated PSTRs in mongo.

Reply:
Added .distinct to map